content/interfaces-and-polymorphism/exercises/_index.md

```diff
@@ -0,0 +1,211 @@
+reviewer: Kimberly Horan # to be set by the page reviewer
+reviewerGitHub: codinglikeagirl42 # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+As a new C# coder, it might take you some time to recognize when to use interfaces.
+
+To help overcome this, let's consider a common occurrence---sorting a
+`List` of objects.
+
+<!-- TODO: Add link to section on list sorting -->
+
+1. If the list contains `string` or numerical entries, then sorting the list
+   is trivial:
+
+   ```csharp
+      listName.Sort();
+   ```
+
+1. However, if the elements are custom objects (like `Cat`), then sorting the
+   list becomes more complicated. This is because the objects may contain
+   multiple fields, any of which could be used as a sorting option. For
+   `Cat` objects, this could include `name`, `age`, or `mass`.
+
+## Getting Started
+
+Open up your [exercises starter code repo](https://github.com/LaunchCodeEducation/csharp-web-dev-exercises) in Visual Studio to get started!
+
+You will practice implementing interfaces by playing around with a small ice
+cream store. It consists of a refrigerated display `Case`, which contains
+a collection of ice cream `Flavor` objects and a selection of `Cone`
+objects.
+
+{{% notice green "Tip" "rocket" %}}
+
+   Did you notice the abstract `Ingredient` class? This gets extended by
+   `Flavor` and `Cone` to help streamline the code.
+
+{{% /notice %}}
+
+## Sorting Flavors by Name
+
+To display a menu for your customers, you need to sort the ice cream flavors
+alphabetically by the `name` field. Fortunately, the `IComparer`
+interface helps you solve the sorting-objects-by-field problem.
+
+{{% notice green "Tip" "rocket" %}}
+
+   Before proceeding, make sure you have read the section on the [IComparer interface]({{< relref "../reading/given-interfaces/#icomparert" >}})!
+
+{{% /notice %}}
+
+### Create a Sorting Class
+
+1. Create a new class called `FlavorComparer` and have it implement the
+   `IComparer` interface:
+
+   ```csharp
+      public class FlavorComparer : IComparer<Flavor>
+   ```
+
+1. To start sorting, we need a `Compare()` method. Add the following code to create one:
+
+   ```csharp {linenos = table}
+      public int Compare(Flavor x, Flavor y)
+      {
+         return string.Compare(x.Name, y.Name);
+      }
+   ```
+
+   This returns an integer (-1, 1, or 0) depending on
+   which `Flavor` object `x` or `y` comes first, alphabetically.
+
+### Sorting the `Flavors` List
+
+In `Program.cs`, we declare `menu` that contains everything in the `Case`
+as well as specific `availableFlavors` and `availableCones` collections.
+
+```csharp {linenos=table, linenostart=10}
+   Case menu = new Case();
+   List<Flavor> availableFlavors = menu.Flavors;
+   List<Cone> availableCones = menu.Cones;
+```
```

Comment:
Suggestion - of tabbing this block of code over to match the ones below for a cleaner look.

Reply:
Thanks for the suggestion! When I tab it over, it turns into a block quote so I am going to leave it for now.